core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -103,6 +104,19 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
       tps.foreach {
         case (tp, offset) => consumer.seek(tp, offset)
       }
+    case AssignOffsetsForTimes(timestampsToSearch) =>
+      scheduleFirstPollTask()
+      checkOverlappingRequests("AssignOffsetsForTimes", sender(), timestampsToSearch.keySet)
+      val previousAssigned = consumer.assignment()
+      consumer.assign((timestampsToSearch.keys.toSeq ++ previousAssigned.asScala).asJava)
+      val topicPartitionToOffsetAndTimestamp = consumer.offsetsForTimes(timestampsToSearch.mapValues(long2Long(_)).asJava)
```

Comment:
note that `offsetsForTimes` is a blocking call, javadoc is even mentioning: "Notice that this method may block indefinitely if the partition does not exist." We already use a dedicated dispatcher for the `KafkaConsumerActor` so it might be alright, but somewhat scary.